packages/provd_client/lib/src/provd_user_client.dart

```diff
@@ -38,8 +39,12 @@ class ProvdUserClient {
 
   /// Validates the [username] and returns true if it is valid.
   Future<bool> validateUsername(String username) async {
-    final request = ValidateUsernameRequest(username: username);
-    final response = await _userClient.validateUsername(request);
-    return response.valid;
+    try {
+      final request = ValidateUsernameRequest(username: username);
+      final response = await _userClient.validateUsername(request);
+      return response.valid;
+    } on GrpcError catch (_) {
```

Comment:
So far provd only returns a boolean here and propagates additional information (i.e. why the username is invalid) as grpc errors. We could either generalize the grpc response (similar to how subiquity does this) or handle the different grpc error cases (once defined) here in the client and then wrap the result in an enum again.

Reply:
We decided to return an enum from the backend instead